hpcflow/tests/unit/test_helper.py

```diff
@@ -0,0 +1,505 @@
+        return_value=Path("/user_data_dir"),
+    )
+    watcher_fp = helper.get_watcher_file_path(app)
+    assert Path("/user_data_dir/watch_workflows.txt") == watcher_fp
+
+
+def test_get_helper_watch_list(app, mocker):
+    mocklist = [{"path": Path("mockpath1")}, {"path": Path("mockpath2")}]
+    mocker.patch(
+        "hpcflow.sdk.helper.helper.get_watcher_file_path",
+        return_value=Path(os.getcwd()),
+    )
+    mocker.patch(
+        "hpcflow.sdk.helper.watcher.MonitorController.parse_watch_workflows_file",
+        return_value=mocklist,
+    )
+    helper_watch_list = helper.get_helper_watch_list(app)
+    assert mocklist == helper_watch_list
+
+
+def test_get_helper_watch_list_no_file(app, mocker):
+    mocklist = [{"path": Path("mockpath1")}, {"path": Path("mockpath2")}]
+    mocker.patch(
+        "hpcflow.sdk.helper.helper.get_watcher_file_path",
+        return_value=Path("/non_existent_watcher_file_path"),
+    )
+    mocker.patch(
+        "hpcflow.sdk.helper.watcher.MonitorController.parse_watch_workflows_file",
+        return_value=mocklist,
+    )
+    helper_watch_list = helper.get_helper_watch_list(app)
+    assert None == helper_watch_list
+
+
+def test_clear_helper_no_process(app):
+    pid_fp = helper.get_PID_file_path(app)
+    with pid_fp.open("wt") as fp:
+        fp.write(f"pid = {12345}\n")
+    helper.clear_helper(app)
+    assert pid_fp.is_file() == False
+
+
+@pytest.mark.skipif(  # Skip MacOs with python 3.7. Also skipping in CentOS at the gh action
+    (sys.platform == "darwin") and (sys.version_info < (3, 8)),
+    reason="Unknown failure. Illegal instruction",
+)
+def test_kill_proc_tree():
+    queue = Queue()
+    depth = 3
+    parent = Process(
+        target=sleeping_child,
+        args=[queue, 10, depth],
+    )
+    parent.start()
+    children = []
+    while len(children) < depth:
+        children.append(queue.get())
+    children.append(parent.pid)
+    try:
+        g, a = helper.kill_proc_tree(parent.pid)
+        assert len(g) == depth + 1
+        assert len(a) == 0
+    finally:
+        sitll_running = 0
+        for pid in children:
+            try:
+                proc = psutil.Process(pid)
+                print(f"Process {proc.pid} still running!")
+                sitll_running = sitll_running + 1
+            except psutil.NoSuchProcess:
+                pass
+        assert sitll_running == 0, "Some processes were not killed"
+
+
+def test_get_helper_uptime(app):
+    # TODO: get_helper_uptime uses proc.create_time, which is known to be inacurate.
+    # See https://github.com/giampaolo/psutil/issues/877
+    # This is probably only an issue for tests, but uptime should not be trusted for
+    # measurements under the "1 second" scale.
+    # When this is solved, the "time.sleep(0.5)" lines can be removed.
+    try:
+        pid_fp = helper.get_PID_file_path(app)
+        t_0 = datetime.now()
+        time.sleep(0.5)
+        proc = subprocess.Popen(["sleep", "100"])
```

Comment:
There's no sleep command in the Windows cmd shell, but there is in powershell: `powershell sleep 100`.